src/Features/Core/Portable/MetadataAsSource/DecompilationMetadataAsSourceFileProvider.cs

```diff
@@ -178,14 +178,18 @@ public DecompilationMetadataAsSourceFileProvider(IImplementationAssemblyLookupSe
                 isReferenceAssembly &&
                 !_implementationAssemblyLookupService.TryFindImplementationAssemblyPath(assemblyLocation, out assemblyLocation))
             {
+                // We didn't find an implementation assembly by any of our heuristic based approaches so
+                // let's see if its in the GAC
                 try
                 {
                     var fullAssemblyName = containingAssembly.Identity.GetDisplayName();
                     GlobalAssemblyCache.Instance.ResolvePartialName(fullAssemblyName, out assemblyLocation, preferredCulture: CultureInfo.CurrentCulture);
                     isReferenceAssembly = assemblyLocation is null;
                 }
-                catch (Exception e) when (FatalError.ReportAndCatch(e, ErrorSeverity.Diagnostic))
+                catch (Exception)
```

Comment:
not a fan of just silent swallow of all exception types.  is there a reason we'e disabling this?  if it's for a known exceptio nthat those apis can throw, can we just silently swallow that instead, and still report other exceptions?

Reply:
looks to be FileLoadException.  Let's just ignore that one then, not all exceptions.

Reply:
You're the boss, so I will update, but I don't get why we'd care about any exception trying to find an implementation assembly in the GAC in this scenario.

Reply:
say we have a null-ref because of our own faulty code (like someone adds a line in here and it's just bogu), i don't want that silently being swallowed leading to a broken expereince.  This helps isolate out: "this failed for a reasonable reason taht can totally happy in teh wild" vs "this failed because we made a bogus mistake and really should own fixing it".

Reply:
I guess I would think that our own faulty code would cause a test failure, so any other issue would be something in the guts of .NET that we don't care about. Unless we're knowingly a canary in the coal mine for these things.